packages/main/src/tray-menu.spec.ts

```diff
@@ -163,3 +163,92 @@ test('Tray click trigger is only added on Windows devices', () => {
   trayMenu = new TrayMenu(tray, animatedTray);
   expect(onSpy).not.toHaveBeenCalledWith('click', expect.any(Function));
 });
+
+test('with a provider already found, should push the addProviderMenuItem fine', async () => {
+  const onSpy = vi.spyOn(ipcMain, 'on');
+  const menuBuild = vi.spyOn(Menu, 'buildFromTemplate');
+
+  trayMenu = new TrayMenu(tray, animatedTray);
+
+  // Add a "provider" so that the menu already finds the provider
+  trayMenu.addProviderItems({
+    id: 'testId',
+    name: 'TestProviderAlreadyHere',
+    internalId: 'internalId',
+  } as ProviderInfo);
+
+  // Do a mock call which should add the menu item (testId matches the provider added above)
+  onSpy.mock.calls[0]?.[1](undefined as unknown as Electron.IpcMainEvent, {
```

Comment:
suggestion: I usually would rather have a check ensuring the mock has been called usually 

```suggestion
  expect(onSpy).toHaveBeenCalledOnce();
  const listener = onSpy.mock.calls[0]?.[1]
  assert(listener);
  
  listener(undefined as unknown as Electron.IpcMainEvent, {
```

Reply:
Otherwise, let's say the onSpy has never been called, you will not get an error at this line, making it harder to debug